### PR TITLE
planner: refactor PrepareExecutor

### PR DESCRIPTION
**Pull request by qw4990**
### What problem does this PR solve?

Issue Number: ref #36598

Problem Summary: planner: refactor PrepareExecutor

### What is changed and how it works?

No logical change, just move the operation of converting `paramSQL` to `PlanCacheStmt` from `PrepareExecutor` to the dedicated static function `GeneratePlanCacheStmt`.

The conversion of `paramSQL -> PlanCacheStmt` is an atomic and general operation for Plan Cache. Making it a function is easier to use and maintain.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

**Review by Yisaer — APPROVED** [review]

File: executor/compiler.go [hunk 1]
```diff
@@ -363,6 +363,10 @@ func getDbFromResultNode(resultNode ast.ResultSetNode) []string { // may have du
 	return dbLabels
 }
 
+func init() {
+	plannercore.GetStmtLabel = GetStmtLabel
```
Comment (hawkingrei) [inline executor/compiler.go:367]: 1、the init of ```plannercore.GetStmtLabel```  should be put at the ```planner/core```. It is direct.

2、``` GetStmtLabel``` should be moved into ```ast```. It is a function of ast. we should reduce the complexity of dependencies.

File: executor/prepared.go [hunk 2]
```diff
@@ -16,28 +16,19 @@ package executor
 
 import (
 	"context"
-	"math"
-
 	"github.com/pingcap/errors"
 	"github.com/pingcap/log"
 	"github.com/pingcap/tidb/expression"
 	"github.com/pingcap/tidb/infoschema"
-	"github.com/pingcap/tidb/parser"
```
Comment (hawkingrei) [inline executor/prepared.go:25]: Please remember to update the bazel config.

File: executor/compiler.go (hunk 1 above)
Comment (qw4990) [reply executor/compiler.go:367]: You are right. I move this function to `ast` now.

File: parser/ast/ast.go [hunk 3]
```diff
@@ -167,3 +167,94 @@ type Visitor interface {
 	// ok returns false to stop visiting.
 	Leave(n Node) (node Node, ok bool)
 }
+
+// GetStmtLabel generates a label for a statement.
+func GetStmtLabel(stmtNode StmtNode) string {
```
Comment (Reminiscent) [inline parser/ast/ast.go:172]: Why move this function to here?

Comment (hawkingrei) [reply parser/ast/ast.go:172]: ```GetStmtLabel``` should be moved into ast. It is a function of ast. we should reduce the complexity of dependencies.

**Review by Reminiscent — APPROVED** [review]

**Review by XuHuaiyu — APPROVED** [review]

**Pull request merged**